### DISCUSSION TITLE
Injector

### DISCUSSION THREAD

**Discussion by KaliStudio** [YimMenu/YimMenu#2438] [General]
I coded a dll injector in C#, the ideal would be to add a function to check for YimMenu updates.

**Comment by Yimura** [comment]
Sorry, releasing binaries like this is not allowed. You can at most link your Github repository here so people can build the source code themselves.

**Reply by KaliStudio** [reply]
The zip contained the source code

**Reply by Yimura** [reply]
Put it on a Github repository, I won't be checking the zip file every 5 seconds to see if it has been changed out with a malicious binary.

And a Github repo is arguably a way better platform.

**Comment by KaliStudio** [comment]
https://github.com/KaliStudio/KaliInjecteur

**Reply by Yimura** [reply]
![image](https://github.com/YimMenu/YimMenu/assets/24669514/9d4e2cbd-74b8-421c-a600-0ea0cfdb09a7)

**Comment by KaliStudio** [comment]
Sorry it's put in private, I just put it public

**Comment by maybegreat48** [comment]
What does [this](https://github.com/KaliStudio/KaliInjecteur/blob/main/Form1.cs#L192C8-L214C1) do?

**Reply by KaliStudio** [reply]
private void Protect()
        {

                // Listes de voyelles et de consonnes possibles
                string[] voyellePossibles = { "a", "e", "i", "o", "u", "y", "au", "aë", "ei", "ou" };
                string[] consonnePossibles = { "a", "b", "c", "d", "f", "g", "h", "j", "k", "l", "m", "n", "p", "q", "r", "s", "t", "v", "ch", "br", "dr", "gr" };

                // Générer un mot aléatoire alternant voyelle-consonne trois fois
                string motAleatoire = GenererMotAleatoire(voyellePossibles, consonnePossibles, 6);

            this.Text = motAleatoire;
        }

**Reply by KaliStudio** [reply]
Random text for a future function that changes the name and title of the app on each startup

**Comment by D-Borkan** [comment]
suggest for this inector
- new style to be more modern
- add buttom to download menu direct and when complete let the injector inject the menu direct in the game

**Reply by KaliStudio** [reply]
Do you have an idea for the style?
I would like to add a system to download and update YimMenu, but for that I would need a way to know the current version and the new version with a download link

**Reply by KaliStudio** [reply]
I updated the code, I'll let you test it.
I need to add a translation system.

**Reply by lonelybud** [reply]
you are also commiting bin and obj files.  I think they kind change frequently.

**Reply by lonelybud** [reply]
> I updated the code, I'll let you test it. I need to add a translation system.

it looks messed up now

**Reply by KaliStudio** [reply]
it downloads Yim if it is not installed, updates it when a new version is available and launches yim when GTA is launched

**Reply by lonelybud** [reply]
you should first format code properly.
 
add .gitignore file - 
```
.vs
bin
obj
```
remove mentioned folder in this file ( git rm -r --cached ). commit .gitignore and those dleted files togther as one commit.

dont auto downlod yimmenu. Provide option to both manually inject from a file location. Also provide a button to check and dowload the latest release, store it. For both individially provide inject button.

use github apis to check and download the dll.

Eg: 

this will get you json
https://api.github.com/repos/YimMenu/YimMenu/releases/latest

this will get you latest dll whatsoever
https://github.com/YimMenu/YimMenu/releases/latest/download/YimMenu.dll

**Reply by D-Borkan** [reply]
so you need to add English language ( or translation system )
- add download buttom and make user know the download is complete then inject
- make injector build with exe just without dll file or ...

another suggest ( may take sometime to do that )
- try to add imgui interface
